spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/SpannerRepositoryImplTests.java

```diff
@@ -0,0 +1,187 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.repository;
+
+import java.util.Arrays;
+
+import com.google.cloud.spanner.Key;
+import com.google.cloud.spanner.KeySet;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import org.springframework.cloud.gcp.data.spanner.core.SpannerOperations;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Chengyuan Zhao
+ */
+@RunWith(SpringRunner.class)
+public class SpannerRepositoryImplTests {
+
+	@Test(expected = IllegalArgumentException.class)
+	public void constructorNullSpannerOperationsTest() {
+		new SpannerRepositoryImpl(null, Object.class);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void constructorNullEntityTypeTest() {
+		new SpannerRepositoryImpl(mock(SpannerOperations.class), null);
+	}
+
+	@Test
+	public void getSpannerOperationsTest() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		assertSame(operations, new SpannerRepositoryImpl(operations, Object.class)
+				.getSpannerOperations());
+	}
+
+	@Test(expected = IllegalArgumentException.class)
```

Comment:
this is great! I think we are missing the same test checks for thrown `IllegalArgumentException` for null argument for the others:
````
	void deleteById(ID id);
	void delete(T entity);
	void deleteAll(Iterable<? extends T> entities);
	<S extends T> Iterable<S> saveAll(Iterable<S> entities);
	Optional<T> findById(ID id);
	boolean existsById(ID id);
````